### PR TITLE
Extract file lock mutex and atomic write utilities

### PR DESCRIPTION
**Pull request by martin-purplefish**
## Summary
- extract cross-process file locking into `src/backend/lib/file-lock-mutex.ts`
- extract atomic file write helper into `src/backend/lib/atomic-file.ts`
- refactor `worktree-lifecycle.service` to use the new lock + atomic write utilities
- refactor `file-lock.service` persistence to use atomic writes
- add focused utility tests and update file-lock service tests for atomic-write behavior

## Why
This separates low-level filesystem safety primitives from domain services, improves reuse, and strengthens test coverage around lock/atomic semantics.

## Testing
- `pnpm vitest src/backend/lib/file-lock-mutex.test.ts src/backend/lib/atomic-file.test.ts src/backend/services/worktree-lifecycle.service.test.ts src/backend/services/file-lock.service.test.ts`
- `pnpm typecheck`

---

> [!NOTE]
> **Medium Risk**
> Refactors cross-process locking and on-disk persistence to new primitives, which can affect concurrency behavior and reliability if edge cases differ from the previous implementations. Risk is mitigated by new focused unit tests around timeout/stale-lock handling and atomic-write cleanup.
> 
> **Overview**
> **Extracts filesystem safety primitives** into reusable utilities: `writeFileAtomic` (temp-file + rename with best-effort cleanup and parent-dir creation) and `FileLockMutex` (lock acquisition with backoff, hard timeout/post-timeout window, and stale-lock removal with inode verification).
> 
> **Refactors services to use the new utilities**: `worktree-lifecycle.service` replaces its inline resume-mode file locking/atomic-write logic with `FileLockMutex` + `writeFileAtomic`, and `file-lock.service` persists `advisory-locks.json` via atomic writes (updating tests to assert temp-path + `rename` usage).
> 
> Adds new Vitest coverage for atomic-write semantics and file-lock mutex behavior (timeouts, post-timeout waiting, stale lock removal, and safe release when lock file is replaced).
> 
> <sup>Written by [Cursor Bugbot](https://cursor.com/dashboard?tab=bugbot) for commit 378be22a8f1b5f4c27831526d1bcd7c81476713e. This will update automatically on new commits. Configure [here](https://cursor.com/dashboard?tab=bugbot).</sup>

### CODE REVIEW

File: src/backend/lib/atomic-file.ts [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+ */
+export async function writeFileAtomic(
+  targetPath: string,
+  content: string | Buffer,
+  options: WriteFileAtomicOptions = {}
+): Promise<void> {
+  const directory = path.dirname(targetPath);
+  await fs.mkdir(directory, { recursive: true });
+
+  const tmpPath = `${targetPath}.${randomUUID()}.tmp`;
```
[the remark this answers is not in the document]
Comment (martin-purplefish) [reply src/backend/lib/atomic-file.ts]: Implemented. `writeFileAtomic()` now creates temp files with a short same-directory prefix (`path.join(dirname(targetPath), `.tmp-<uuid>`)`) instead of embedding the full target filename. This preserves same-directory atomic rename semantics while avoiding long/leaky temp names. Updated related assertions in `src/backend/services/file-lock.service.test.ts` accordingly.

File: src/backend/lib/file-lock-mutex.ts [hunk 2]
```diff
@@ -0,0 +1,244 @@
...
+      initialRetryDelayMs: options.initialRetryDelayMs ?? DEFAULT_LOCK_RETRY_DELAY_MS,
+      maxRetryDelayMs: options.maxRetryDelayMs ?? DEFAULT_LOCK_MAX_RETRY_DELAY_MS,
+      maxStaleRetries: options.maxStaleRetries ?? DEFAULT_LOCK_MAX_STALE_RETRIES,
+      staleThresholdMs: options.staleThresholdMs ?? acquireTimeoutMs * 5,
+    };
+  }
+
+  async acquire(lockPath: string): Promise<() => Promise<void>> {
+    const startTime = Date.now();
+    let retryDelay = this.options.initialRetryDelayMs;
```
[the remark this answers is not in the document]
Comment (martin-purplefish) [reply src/backend/lib/file-lock-mutex.ts]: Implemented. `FileLockMutex.acquire()` now uses a bounded post-timeout grace window (`postTimeoutWaitMs`) before failing, so it continues retrying under contention after `acquireTimeoutMs` instead of throwing immediately. Stale-lock cleanup is still attempted after the initial timeout, and timeout now occurs at `acquireTimeoutMs + postTimeoutWaitMs`. Added a regression test: `keeps waiting during post-timeout window and acquires once lock is released` in `src/backend/lib/file-lock-mutex.test.ts`.
